### PR TITLE
SWC-7164: SRC FullWidthAlert updates

### PR DESCRIPTION
**Pull request by kianamcc**
Jira ticket: https://sagebionetworks.jira.com/browse/SWC-7164

Screenshot 2024-12-11 at 12 31 53 PM

### CODE REVIEW

File: packages/synapse-react-client/src/components/FullWidthAlert/FullWidthAlert.tsx [hunk 1]
```diff
@@ -102,6 +107,9 @@ function FullWidthAlert(props: FullWidthAlertProps) {
     globalAlertSx,
   } = props
 
+  const theme = useTheme()
+  const isMobile = useMediaQuery(theme.breakpoints.down('sm'))
```
Comment (kianamcc) [inline packages/synapse-react-client/src/components/FullWidthAlert/FullWidthAlert.tsx:111]: I use useMediaQuery to handle code with conditional rendering

File: packages/synapse-react-client/src/components/FullWidthAlert/FullWidthAlert.tsx [hunk 2]
```diff
@@ -133,26 +144,43 @@ function FullWidthAlert(props: FullWidthAlertProps) {
...
+            <>
+              {title && (
+                <AlertTitle
+                  sx={{ margin: 0, textAlign: { xs: 'center', sm: 'initial' } }}
+                >
+                  {title}
+                </AlertTitle>
+              )}
+            </>
+          ) : (
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/FullWidthAlert/FullWidthAlert.tsx:162]: This is a bad idea, not everywhere that we use FullWidthAlert do we guarantee that we show a title, or that the title/description by itself is enough for the user to understand what's going on.

Let's revert this part of the change and decide if we should change only the instance of the alert shown in the design, or if we should do a comprehensive review of all places where we use this component.

File: packages/synapse-react-client/src/components/FullWidthAlert/FullWidthAlert.tsx [hunk 3]
```diff
@@ -133,26 +144,43 @@ function FullWidthAlert(props: FullWidthAlertProps) {
...
+            direction={{ xs: 'row', sm: 'column', lg: 'row' }}
             alignItems="center"
             display="flex"
             flexShrink={0}
+            sx={{
+              gap: { xs: '8px', sm: 'initial' },
+              flexWrap: { xs: 'wrap', sm: 'nowrap' },
+              justifyContent: { xs: 'center', sm: 'initial' },
+              width: { xs: '100%', sm: 'initial' },
+            }}
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/FullWidthAlert/FullWidthAlert.tsx:183]: These seem ok, but I want to try them out in a story to verify

**Review by nickgros — CHANGES_REQUESTED** [review]

File: packages/synapse-react-client/src/components/FullWidthAlert/FullWidthAlert.tsx [hunk 4]
```diff
@@ -145,14 +152,15 @@ function FullWidthAlert(props: FullWidthAlertProps) {
...
-              lg: 'row',
-            }}
+            direction={{ xs: 'row', sm: 'column', lg: 'row' }}
             alignItems="center"
             display="flex"
             flexShrink={0}
+            sx={{
+              gap: { xs: '8px', sm: 'initial' },
+              flexWrap: { xs: 'wrap', sm: 'nowrap' },
+              width: { xs: '100%', sm: 'initial' },
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/FullWidthAlert/FullWidthAlert.tsx:162]: What do you think of adding this to make the buttons expand to full width at small screen sizes?
```suggestion
              width: { xs: '100%', sm: 'initial' },
              [theme.breakpoints.down('xs')]: {
                ["& > *"]: {
                  flexGrow: 1
                }
              }
```

File: packages/synapse-react-client/src/components/FullWidthAlert/FullWidthAlert.tsx [hunk 5]
```diff
@@ -145,14 +152,15 @@ function FullWidthAlert(props: FullWidthAlertProps) {
...
-            direction={{
-              xs: 'column-reverse',
-              sm: 'column',
-              lg: 'row',
-            }}
+            direction={{ xs: 'row', sm: 'column', lg: 'row' }}
             alignItems="center"
             display="flex"
             flexShrink={0}
+            sx={{
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/FullWidthAlert/FullWidthAlert.tsx:159]:
```suggestion
            sx={theme => ({
```

File: packages/synapse-react-client/src/components/FullWidthAlert/FullWidthAlert.tsx [hunk 6]
```diff
@@ -145,14 +152,15 @@ function FullWidthAlert(props: FullWidthAlertProps) {
...
+              gap: { xs: '8px', sm: 'initial' },
+              flexWrap: { xs: 'wrap', sm: 'nowrap' },
+              width: { xs: '100%', sm: 'initial' },
+            }}
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/FullWidthAlert/FullWidthAlert.tsx:163]:
```suggestion
            })}
```

**Review by nickgros — APPROVED** [review]

**Pull request merged**